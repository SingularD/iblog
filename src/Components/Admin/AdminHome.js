import React from 'react'

class AdminHome extends React.Component{
  constructor(props) {
    super(props)

  }

  render() {
    return (
      <div className="admin-home-container">
        <h1>AdminHome</h1>
      </div>
    )
  }
}

export default AdminHome
