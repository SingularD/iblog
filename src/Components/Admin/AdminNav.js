import React from 'react'

class AdminNav extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="admin-nav-container col-2 row justify-content-center bg-light">
        <div className="col-10">
          <h2>后台管理</h2>
          <ul className="list-group admin-nav-link">
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
            <li className="list-group-item border-0">首页</li>
          </ul>
        </div>
      </div>
    )
  }

}

export default AdminNav
