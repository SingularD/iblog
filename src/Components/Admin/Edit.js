import React from 'react'
import { Link } from "react-router-dom";
import { editData } from "../../Data";

class Edit extends React.Component{
  constructor(props) {
    super(props)
    this.state = {
      newCategory: '',
      newResume: '',
      category: editData.category,
      postList: editData.postList
    }
    this.getNewCategory = this.getNewCategory.bind(this)
    this.addCategoryItem = this.addCategoryItem.bind(this)
    this.getNewResume = this.getNewResume.bind(this)
  }
  getNewCategory(e) {
    this.setState({
      newCategory: e.target.value
    })
  }
  getNewResume(e) {
    this.setState({
      newResume: e.target.value
    })
  }
  addCategoryItem() {
    let temp = {
      categoryTitle: this.state.newCategory,
      resume: this.state.newResume
    }
    const arr = [...this.state.category, temp]
    this.setState({
      category: arr
    })
  }
  render() {
    return (
      <div className="edit col-10 justify-content-center row">
        <div className="col-10">
          <div>
            <h2>专题标签编辑：</h2>
            <div className="row justify-content-between">
              <div className="col">
                <ul className="list-group category-group">
                  {
                    this.state.category.map((item, index) => (
                      <li key={index}>
                        <p>
                          {item.categoryTitle}
                        </p>
                        <p>
                          {item.resume}
                        </p>
                      </li>
                    ))
                  }
                </ul>
              </div>
              <div className="col">
                <div className="input-group mb-3 row">
                  <div className="col-12">
                    <input
                      type="text"
                      className="form-control"
                      placeholder="输入新专题名"
                      onChange={this.getNewCategory}
                    />
                  </div>
                    <div className="col-12 pt-5">
                      <textarea
                        className="form-control"
                        placeholder="请输入该标题相关简介"
                        onChange={this.getNewResume}
                      >
                      </textarea>
                    </div>
                  <div className="col-12 row justify-content-center pt-5">
                    <button
                      className="btn btn-primary col-4"
                      onClick={this.addCategoryItem}
                    >
                      提交
                    </button>
                  </div>
                </div>

              </div>
            </div>
          </div>
          <div>
            <h2>文章编辑：</h2>
            <ul className="list-group list-group-flush post-group">
              {
                this.state.postList.map((item, index) => (
                  <li key={index}>
                    <div className="row justify-content-between">
                      <div className="passage-title col-8">
                        <p>{item.title}({item.date})</p>
                      </div>
                      <div className="passage-edit col-2 row justify-content-between">
                        <Link to="/admin/edit-item/123">
                          编辑
                        </Link>
                        <Link to="/admin/edit-item/">
                          删除
                        </Link>
                      </div>
                    </div>
                  </li>
                ))
              }
            </ul>
          </div>
        </div>
      </div>
    )
  }

}

export default Edit
