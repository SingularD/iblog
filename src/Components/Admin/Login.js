import React from 'react'

class Login extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="login-container">
        <div className="login-input-group">
          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text d-none d-lg-block" id="basic-addon1">Username</span>
            </div>
            <input type="text" className="form-control" placeholder="Username" />
          </div>
          <div className="input-group mb-3">
            <div className="input-group-prepend">
              <span className="input-group-text d-none d-lg-block" id="basic-addon1">Password</span>
            </div>
            <input type="password" className="form-control" placeholder="Password" />
          </div>
          <div>
            <button className="btn btn-primary">登录</button>
          </div>
        </div>
      </div>
    )
  }

}

export default Login
