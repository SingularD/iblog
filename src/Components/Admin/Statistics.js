import React from 'react'
import { chartData, RadarData, BarData } from "../../Data";
import {AreaChart, XAxis, YAxis, CartesianGrid, Tooltip, Area, BarChart, Bar,
  RadarChart, PolarGrid, PolarAngleAxis, PolarRadiusAxis, Radar, Legend
} from "recharts";


class Statistics extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="statistics col-10 justify-content-center row">
        <div className="pv-chart col-10 pt-5">
          <h1>最近7日PV/UV统计</h1>
          <AreaChart width={730} height={250} data={chartData}
                     margin={{ top: 10, right: 30, left: 0, bottom: 0 }}>
            <defs>
              <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
                <stop offset="5%" stopColor="#8884d8" stopOpacity={0.8}/>
                <stop offset="95%" stopColor="#8884d8" stopOpacity={0}/>
              </linearGradient>
              <linearGradient id="colorPv" x1="0" y1="0" x2="0" y2="1">
                <stop offset="5%" stopColor="#82ca9d" stopOpacity={0.8}/>
                <stop offset="95%" stopColor="#82ca9d" stopOpacity={0}/>
              </linearGradient>
            </defs>
            <XAxis dataKey="name" />
            <YAxis />
            <CartesianGrid strokeDasharray="3 3" />
            <Tooltip />
            <Area type="monotone" dataKey="uv" stroke="#8884d8" fillOpacity={1} fill="url(#colorUv)" />
            <Area type="monotone" dataKey="pv" stroke="#82ca9d" fillOpacity={1} fill="url(#colorPv)" />
          </AreaChart>
          <h1 className="pt-5">文章PV/UV</h1>
          <BarChart width={730} height={250} data={BarData}>
            <CartesianGrid strokeDasharray="3 3" />
            <XAxis dataKey="name" />
            <YAxis />
            <Tooltip />
            <Legend />
            <Bar dataKey="pv" fill="#8884d8" />
            <Bar dataKey="uv" fill="#82ca9d" />
          </BarChart>
        </div>
        <div className="pv-chart col-10 pb-5">
          <h1>专题统计</h1>
          <RadarChart outerRadius={90} width={730} height={250} data={RadarData}>
            <PolarGrid />
            <PolarAngleAxis dataKey="subject" />
            <PolarRadiusAxis angle={30} domain={[0, 150]} />
            <Radar name="Mike" dataKey="A" stroke="#8884d8" fill="#8884d8" fillOpacity={0.6} />
            <Radar name="Lily" dataKey="B" stroke="#82ca9d" fill="#82ca9d" fillOpacity={0.6} />
            <Legend />
          </RadarChart>
          <h1>归档统计</h1>
          <RadarChart outerRadius={90} width={730} height={250} data={RadarData}>
            <PolarGrid />
            <PolarAngleAxis dataKey="subject" />
            <PolarRadiusAxis angle={30} domain={[0, 150]} />
            <Radar name="Mike" dataKey="A" stroke="#8884d8" fill="#8884d8" fillOpacity={0.6} />
            <Radar name="Lily" dataKey="B" stroke="#82ca9d" fill="#82ca9d" fillOpacity={0.6} />
            <Legend />
          </RadarChart>
        </div>
      </div>
    )
  }
}

export default Statistics
