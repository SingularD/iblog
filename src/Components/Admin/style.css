.admin-nav-link {
    padding-top: 60px;
}
.admin-nav-link li{
    padding-top: 20px;
    padding-left: 0 !important;
    padding-right: 0 !important;
}
.statistics {
    height: 100vh;
    overflow: auto;
}
.writing {
    height: 100vh;
    overflow: auto;
}
.edit {
    height: 100vh;
    overflow: auto;
}
