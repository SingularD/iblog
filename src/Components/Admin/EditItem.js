import React from 'react'
import './style.css'

class EditItem extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <div className="writing col-10 justify-content-center row">
        <div className="col-10 writing-content pt-5">
          <div className="form-group">
            <label htmlFor="exampleFormControlInput1">文章标题</label>
            <input type="email" className="form-control" id="exampleFormControlInput1" placeholder="文章标题"/>
          </div>
          <div className="form-group">
            <label htmlFor="exampleFormControlSelect1">文章专题</label>
            <select className="form-control" id="exampleFormControlSelect1">
              <option>HTTP</option>
              <option>JAVA</option>
              <option>ReactJs</option>
              <option>Spring Boot</option>
              <option>其它</option>
            </select>
          </div>
          <div className="form-group">
            <label htmlFor="exampleFormControlTextarea1">文章简介</label>
            <textarea className="form-control" id="exampleFormControlTextarea1" rows="1" placeholder="文章简介"></textarea>
          </div>
          <div className="form-group">
            <label htmlFor="exampleFormControlTextarea1">文章内容</label>
            <textarea className="form-control" id="exampleFormControlTextarea1" rows="3" placeholder="文章标题"></textarea>
          </div>
          <div className="writing-preview">
            <h2>预览:</h2>

          </div>
          <div className="row justify-content-around">
            <button className="btn btn-primary col-3">预览</button>
            <button className="btn btn-success col-3">提交</button>
          </div>
        </div>
      </div>
    )
  }

}

export default EditItem
