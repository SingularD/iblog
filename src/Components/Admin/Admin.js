import React from 'react'
import {BrowserRouter, Route, Switch, Redirect} from "react-router-dom";
import './style.css'


import AdminNav from './AdminNav'
import Statistics from './Statistics'
import Writing from './Writing'
import Edit from './Edit'
import EditItem from './EditItem'

class Admin extends React.Component{
  constructor(props) {
    super(props)
    this.state = {
      login: ''
    }
  }
  componentWillMount() {
    this.setState({
      login: localStorage.getItem('iblog_login')
    })
  }

  render() {
    return (
      <React.Fragment>
      {
        this.state.login === 'true' ?
            <BrowserRouter>
              <AdminNav />
              <Switch>
                <Route path="/admin/edit" component={Edit}/>
                <Route path="/admin/writing" component={Writing}/>
                <Route path="/admin/edit-item/:id" component={EditItem} />
                <Route path="/admin/" component={Statistics}/>
              </Switch>
            </BrowserRouter>
           :
          <Redirect to="/login"/>
      }
      </React.Fragment>
    )
  }

}

export default Admin
