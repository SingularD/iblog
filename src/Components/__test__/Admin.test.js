import React from 'react'
import { shallow } from "enzyme";
import Admin from '../Admin/Admin'

describe("检测后台管理页面是否正常渲染", () => {
  let wrapper = shallow(<Admin />)
  it("快照匹配", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
