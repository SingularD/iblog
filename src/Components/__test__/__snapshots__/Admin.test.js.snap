// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`检测后台管理页面是否正常渲染 快照匹配 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Admin />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Redirect
        to="/login"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "to": "/login",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Redirect
          to="/login"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "to": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
